### ISSUE TITLE
[Bug]: focus state doesn't always have a reasonable place to go causing a11y violations

### ISSUE THREAD

**Issue by ethanwinters** [carbon-design-system/carbon-ai-chat#59]
Currently, the default focus target for the chat is the input field. If something causes the input field to be disabled, the focus will fallback to the "close" button if it doesn't know where to go. If there is no close button, it will focus on the message list.

We should:

1. Allow the input field to receive focus if its temporarily disabled (not removed, disabled)
2. We should put focus on the message list instead of the close button if there is no input field to go to
